js/stores/windowStore.js

```diff
@@ -142,14 +142,15 @@ const doAction = (action) => {
       break
     case WindowConstants.WINDOW_SET_LOCATION:
       const key = action.key || windowState.get('activeFrameKey')
-      const lastLocation = windowState.getIn(frameStatePath(key).concat(['location']))
+      const nextLocation = action.location.split('?')[0].split('#')[0]
```

Comment:
won't this pull the named anchor from the wrong place? http://tools.ietf.org/html/rfc3986#section-4.1
Having it before the query string might be non-standard behavior that we also have to support, but either way it's probably better to use URL.parse/format